cpp/tests/utilities/base_fixture.hpp

```diff
@@ -176,50 +178,30 @@ class RmmTestEnvironment : public ::testing::Environment {
 /**---------------------------------------------------------------------------*
  * @brief String representing which RMM allocation mode is to be used.
  *
- * Valid values are 'cuda', 'pool' and 'managed'. 
+ * 
  *---------------------------------------------------------------------------**/
-  std::string mode;
+  std::unique_ptr<rmm::mr::device_memory_resource> rmm_resource{};
 public:
   /**
-   * @brief Constructor where the only argument is a string `mode`.
-   *
-   * Mode repesents the rmm allocation mode, @ref mode.
-   */
-  RmmTestEnvironment(std::string const& mode):mode(mode){}
-
-  /**
-   * @brief Initializes the default RMM memory resource.
-   *
-   * Sets different allocation mode based on the value passed to the constructor.
+   * @brief Sets the default RMM memory resource based on the input string.
    * 
-   * @throws cudf::logic_error if mode value is invalid.
+   * @param allocation_mode Represents the type of the memory resource to be 
+   * used as a default resource. Valid values are 'cuda', 'pool' and 'managed'.
    * 
-   * @return void
+   * @throws cudf::logic_error if passed mode value is invalid.
    */
-  void SetUp() {
-    rmmOptions_t opts{};
-    if (mode == "cuda")
-      opts.allocation_mode = CudaDefaultAllocation;
-    else if (mode == "pool")
-      opts.allocation_mode = PoolAllocation;
-    else if (mode == "managed")
-      opts.allocation_mode = CudaManagedMemory;
+  RmmTestEnvironment(std::string const& allocation_mode) {
```

Comment:
I think the CamelCase is leftover from before we had a standard. Shouldn't this be `rmm_test_environment`?